iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2174,6 +2176,62 @@ static STRING_HANDLE buildClientId(const char* device_id, const char* module_id)
     }
 }
 
+static int append_optional_connect_parameters(PMQTTTRANSPORT_HANDLE_DATA transport_data)
+{
+    int result = 0;
```

Comment:
>int result = 0 [](start = 4, length = 14)

Please leave result as uninitialized.  This will make it so compiler will catch any places that we don't set this.

This would mean we'd need to set result=MU_FAILURE explicitly in the various cases like out-of-memory.